### DISCUSSION TITLE
twitch chat login via chromium parameters not working

### DISCUSSION THREAD

**Discussion by Wandang** [streamlink/streamlink-twitch-gui#832] [Q&A] [answered]
I would like to login to the chat automatically but this never works for me.

What works: starting a stream with chat and using traditional user:pw combo. This needs to be done whenever a stream is started/restarted,

What I would like to have: Using the user:pw combo or a token to always login to chat automatically

What I tried:
adding parameter to chromium for chat in various variations:

* `"--user={<username>}" "--token={<token>}"`
* `--user="{<username>}" --token="{<token>}"
`

I used the token that is already linked in streamlink-twitch-gui (via profile menu)
used a newly generated token via https://twitchtokengenerator.com. unfortunately all of the required scopes from the package.json are deprecated. So I used these new ones instead:

* chat:read
* chat:edit

Nothing worked. What am I doing wrong?

All the best,
Wandang

**Comment by bastimeyer** [comment, marked as answer]
> What am I doing wrong?

Chromium uses the cookie data of the main Twitch website to log into Twitch chat. It's just a regular browser instance/session with the Twitch website loaded, but with a minimal window layout ("Chromium app mode"). You can't pass parameters here like applications which are specifically built for Twitch chat, like Chatty or Chatterino. This doesn't make sense.

If you want to be logged in while using the Chromium chat profile, then log in on the Twitch website in your regular Chromium session, so that Chromium can pick up the cookie data when it's launched from the Twitch GUI in the app mode. If you want to have a separate Chromium session, then use the custom `--user-data-dir=PATH` parameter.

**Reply by Wandang** [reply]
Thanks for the fast response Basti!

In that case the GUI is kinda confusing since it explicitly shows chromium parameters(+ token as an param example) if you select chromium. Maybe this option should be removed in that case?

Based on your suggestion I removed the chromium setting of clearing session and cookie data on window close and did login there normally. Now I can use this session normally for each reopened chat.

This is awesome. thanks so much

**Reply by bastimeyer** [reply]
It's showing parameters for all presets, regardless of what you've chosen. Technically, you can still use all the available variables in the Chromium preset's custom parameter string. Maybe someone has a use for this as there are a ton of parameters which Chromium has, but logging in with the OAuth token is not possible, as you can't set cookies from Chromium's CLI AFAIA. Removing custom parameters or a subset of them for specific profiles would require additional logic and would introduce unnecessary restrictions.
